franklin/github/views.py

```diff
@@ -209,8 +148,24 @@ def repository_detail(request, pk):
         serializer = SiteSerializer(site)
         return Response(serializer.data)
     elif request.method == 'DELETE':
-        site.delete()
-        return Response(status=status.HTTP_204_NO_CONTENT)
+        site.is_active = False
+        site.save()
+        return delete_site(site, request.user)
+
+
+def delete_site(site, user):
+    # TODO - This should be an async process that is thrown into a queue. The
+    # site is no longer active, so the actual delete can occur lazily.
+    webhook_delete_response = delete_webhook(site, user)
+    if status.is_success(webhook_delete_response.status_code):
+        deploy_key_delete_response = delete_deploy_key(site, user)
+        if status.is_success(deploy_key_delete_response.status_code):
+            site.delete()
+            return Response(status=status.HTTP_204_NO_CONTENT)
+        else:
+            message = 'Github deleted the webhook, but not the deploy key'
+            logger.warn(message + ' | %s | %s', user, site)
+    return Response(status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Right now this delete logic is called from 2 places above. One of the calls ignores the returns. Again...this will get refactored.